src/playground/playground.js

```diff
@@ -40,6 +40,57 @@ const getAssetUrl = function (asset) {
     return assetUrlParts.join('');
 };
 
+const addProfilerPanels = function (
+    stats, vm, panelName = 'BLK%', panelFg = '#fff', panelBg = '#111') {
+
+    vm.runtime.enableProfiling();
+    if (vm.runtime.profiler === null) {
+        // Profiler isn't available on the local system.
+        return;
+    }
+
+    const blockPercentPanel = stats.addPanel(new window.Stats.Panel(panelName, panelFg, panelBg));
+
+    const profiler = vm.runtime.profiler;
+    vm.runtime.profiler = null;
+
+    const stepThreadsProfilerId = profiler.idByName('Sequencer.stepThreads');
+    const blockFunctionProfilerId = profiler.idByName('blockFunction');
+
+    let executeTime = 0;
+    const stepValues = [];
+    let lastUpdate = Date.now();
+
+    profiler.onFrame = function ({id, totalTime}) {
+        if (id === stepThreadsProfilerId && totalTime > 0) {
+            stepValues.push(executeTime / totalTime * 100);
+            if (Date.now() - lastUpdate > 1000) {
+                lastUpdate = Date.now();
+                const average = stepValues.reduce(
+                    (a, b) => a + b,
+                    0) / stepValues.length;
+                blockPercentPanel.update(average, 100);
+                stepValues.length = 0;
+            }
+            executeTime = 0;
+        } else if (id === blockFunctionProfilerId) {
+            executeTime += totalTime;
+        }
+    };
+
+    blockPercentPanel.dom.style.display = 'none';
+
+    stats.dom.addEventListener('click', () => {
+        if (blockPercentPanel.dom.style.display === 'block') {
+            vm.runtime.profiler = profiler;
+        } else {
+            vm.runtime.profiler = null;
```

Comment:
I also don't understand why the profiler is being set to null here...

Reply:
This if/else block is checking if the Block % stats panel is visible or not. If its visible `=== 'block'`, enable profiling, which will call onFrame, which will update the stats panel every second onFrame is called. If `!== 'block'` it sets the profiler to null, disabling the profiling.

That way the profiler records when that stats panel is visible, and doesn't record when it is not visible.